src/topology/uniform_space/uniform_convergence.lean

```diff
@@ -0,0 +1,334 @@
+`f` if, for any entourage of the diagonal `u`, for any `x ∈ s`, one has eventually `(f x, Fₙ x) ∈ u`
+for all `y` in a neighborhood of `x` in `s`. -/
+def tendsto_locally_uniformly_on (F : ι → α → β) (f : α → β) (p : filter ι) (s : set α) :=
+  ∀ u ∈ 𝓤 β, ∀ x ∈ s, ∃ t ∈ nhds_within x s, ∀ᶠ n in p, ∀ y ∈ t, (f y, F n y) ∈ u
+
+/-- A sequence of functions `Fₙ` converges locally uniformly to a limiting function `f` if, for any
+entourage of the diagonal `u`, for any `x`, one has eventually `(f x, Fₙ x) ∈ u`
+for all `y` in a neighborhood of `x`. -/
+def tendsto_locally_uniformly (F : ι → α → β) (f : α → β) (p : filter ι) :=
+  ∀ u ∈ 𝓤 β, ∀ (x : α), ∃ t ∈ 𝓝 x, ∀ᶠ n in p, ∀ y ∈ t, (f y, F n y) ∈ u
+
+lemma tendsto_uniformly_on.tendsto_locally_uniformly_on
+  (h : tendsto_uniformly_on F f p s) : tendsto_locally_uniformly_on F f p s :=
+λ u hu x hx, ⟨s, self_mem_nhds_within, h u hu⟩
+
+lemma tendsto_uniformly.tendsto_locally_uniformly
+  (h : tendsto_uniformly F f p) : tendsto_locally_uniformly F f p :=
+λ u hu x, ⟨univ, univ_mem_sets, by simpa using h u hu⟩
+
+lemma tendsto_locally_uniformly_on.mono (h : tendsto_locally_uniformly_on F f p s) (h' : s' ⊆ s) :
+  tendsto_locally_uniformly_on F f p s' :=
+begin
+  assume u hu x hx,
+  rcases h u hu x (h' hx) with ⟨t, ht, H⟩,
+  exact ⟨t, nhds_within_mono x h' ht, H.mono (λ n, id)⟩
+end
+
+lemma tendsto_locally_uniformly_on_univ :
+  tendsto_locally_uniformly_on F f p univ ↔ tendsto_locally_uniformly F f p :=
+by simp [tendsto_locally_uniformly_on, tendsto_locally_uniformly, nhds_within_univ]
+
+lemma tendsto_locally_uniformly_on.comp [topological_space γ] {t : set γ}
+  (h : tendsto_locally_uniformly_on F f p s)
+  (g : γ → α) (hg : maps_to g t s) (cg : continuous_on g t) :
+  tendsto_locally_uniformly_on (λ n, (F n) ∘ g) (f ∘ g) p t :=
+begin
+  assume u hu x hx,
+  rcases h u hu (g x) (hg hx) with ⟨a, ha, H⟩,
+  have : g⁻¹' a ∈ nhds_within x t :=
+    ((cg x hx).preimage_mem_nhds_within' (nhds_within_mono (g x) hg.image_subset ha)),
+  exact ⟨g ⁻¹' a, this, H.mono (λ n hn y hy, hn _ hy)⟩
+end
+
+lemma tendsto_locally_uniformly.comp [topological_space γ]
+  (h : tendsto_locally_uniformly F f p) (g : γ → α) (cg : continuous g) :
+  tendsto_locally_uniformly (λ n, (F n) ∘ g) (f ∘ g) p :=
+begin
+  rw ← tendsto_locally_uniformly_on_univ at h ⊢,
+  rw continuous_iff_continuous_on_univ at cg,
+  exact h.comp _ (maps_to_univ _ _) cg
+end
+
+/-!
+### Uniform approximation
+
+In this section, we give lemmas ensuring that a function is continuous if it can be approximated
+uniformly by continuous functions. We give various versions, within a set or the whole space, at
+a single point or at all points, with locally uniform approximation or uniform approximation. All
+the statements are derived from a statement about locally uniform approximation within a set at
+a point, called `continuous_within_at_of_locally_uniform_approx_of_continuous_within_at`. -/
```

Comment:
Should we have a definition for this `L` assumption?

Reply:
I am not sure. I don't think this will show up often (although I have stated the lemmas in this version because it seems to be the minimal version where it holds, and one never knows if this will be needed some day). I would say let's keep it like that, and if it turns out that people use this version then we can add a name.